src/sonic-yang-models/yang-models/sonic-device_metadata.yang

```diff
@@ -225,6 +225,15 @@ module sonic-device_metadata {
                     }
                     description "Information of rack mgmt map.";
                 }
+
+                leaf timezone {
+                    type stypes:timezone-name-type {
+                        length 1..255;
+                    }
```

Comment:
Why not move length to sonic-types.yang?

Reply:
`timezone-name-type` was copied from the Openconfig. I prefer to leave it unchanged there. So that is why I add the length definition here.

Reply:
Do you really need length?

Reply:
Yes, we shouldn't allow empty timezones, as well as enormous long strings, like more than 255 characters long.